src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassReportServiceImpl.java

```diff
@@ -81,56 +86,111 @@ public String getFileContents(String fileName) {
     }
 
     @Override
-    public void writeBatchJobReport(RassBatchJobReport rassBatchJobReport) {
-        LOG.info("writeBatchJobReport: start writing RASS batch job report");
-        reportWriterService.initialize();
+    public void writeBatchJobReports(RassBatchJobReport rassBatchJobReport) {
+        LOG.info("writeBatchJobReports: start writing RASS batch job reports for each file");
+
         List<RassXmlFileParseResult> fileParseResults = rassBatchJobReport.getFileParseResults();
+        Map<String, RassXmlFileProcessingResult> fileProcessingResults = rassBatchJobReport.getFileProcessingResults();
+        List<RassXmlFileParseResult> errorResults = new ArrayList<>();
 
         for (RassXmlFileParseResult fileParseResult : fileParseResults) {
             if (RassConstants.RassParseResultCode.ERROR.equals(fileParseResult.getResultCode())) {
-                reportWriterService.writeFormattedMessageLine("RASS file: " + fileParseResult.getRassXmlFileName() + " hasn't been processed successfully");
-            } else {
-                reportWriterService.writeFormattedMessageLine("RASS file: " + fileParseResult.getRassXmlFileName() + " has been processed successfully");
-
-                writeReportSummary(rassBatchJobReport);
-                writeReportDetails(rassBatchJobReport);
+                errorResults.add(fileParseResult);
+                continue;
             }
+            RassXmlFileProcessingResult fileResult = fileProcessingResults.get(fileParseResult.getRassXmlFileName());
+            initializeNewReportForFile(fileParseResult);
+            writeReportSummary(fileResult);
+            writeReportDetails(fileResult);
+            finalizeReportForCurrentFile();
+            sendReportEmail();
         }
-        reportWriterService.destroy();
-        LOG.info("writeBatchJobReport: finished writing RASS batch job report");
+        
+        if (!errorResults.isEmpty()) {
+            LOG.info("writeBatchJobReports: One or more RASS files failed to be parsed; writing error report file");
+            writeErrorReport(errorResults);
+            sendReportEmail();
+        }
+        
+        LOG.info("writeBatchJobReports: finished writing RASS batch job reports");
+    }
+
+    private void writeErrorReport(List<RassXmlFileParseResult> errorResults) {
+        initializeNewReport(RassConstants.RASS_PARSE_ERRORS_BASE_FILENAME);
+        reportWriterService.writeFormattedMessageLine("RASS Error Report");
+        reportWriterService.writeNewLines(1);
+        reportWriterService.writeFormattedMessageLine(
+                "Unexpected errors were encountered when attempting to read the following RASS XML files.");
+        reportWriterService.writeFormattedMessageLine(
+                "The failures were likely the result of incorrect XML formatting.");
+        reportWriterService.writeFormattedMessageLine(
+                "Specific details are available in the RASS batch job logs.");
```

Comment:
Should these three lines be moved to a property or a parameter?

Reply:
Ok, I'll get these three strings externalized accordingly.